python/ray/tune/trial.py

```diff
@@ -773,7 +773,11 @@ def __setstate__(self, state):
         for key in self._nonjson_fields:
             state[key] = cloudpickle.loads(hex_to_binary(state[key]))
 
+        # Ensure that stub doesn't get overriden
+        stub = getattr(self, "stub", None)
         self.__dict__.update(state)
+        self.stub = stub if stub is not None else self.stub
```

Comment:
Actually should this logic even be here? Should this be generic to all `Trial`s? Why not just contain it in `make_stub_if_needed`?

Reply:
This should be generic to all trials, yes. See `load_trials_from_experiment_checkpoint`

Reply:
The idea here was to be able to override the stub attribute by setting it before calling `__setstate__`, otherwise to use the stub that was set in the loaded state.

Reply:
I think we can leave this out here (remove this part of the code completely), right?